data_steward/cdr_cleaner/cleaning_rules/remove_non_matching_participant.py

```diff
@@ -35,214 +45,275 @@
 (
-SELECT 
+  SELECT 
     *,
-    ({key_fields_criteria}) AS key_fields_criteria,
-    ({all_fields_criteria}) AS all_fields_criteria
-FROM `{project_id}.{validation_dataset_id}.{identity_match_table}`
+    ({{key_fields_criteria}}) AS key_fields_criteria,
+    ({{all_fields_criteria}}) AS all_fields_criteria
+  FROM `{{project_id}}.{{validation_dataset_id}}.{{identity_match_table}}`
 )
-
-SELECT
-    person_id
+SELECT person_id
 FROM non_match_participants
-WHERE key_fields_criteria IS TRUE OR all_fields_criteria IS TRUE
-"""
+WHERE (key_fields_criteria IS TRUE OR all_fields_criteria IS TRUE)
+{{not_validated_participants_expr}}
+""")
 
+NOT_VALIDATED_PARTICIPANTS_TEMPLATE = JINJA_ENV.from_string("""
+AND person_id IN ({{pids}})                                                            
+""")
 
-def exist_participant_match(ehr_dataset_id, hpo_id):
-    """
-    This function checks if the hpo has submitted the participant_match data 
-    
-    :param ehr_dataset_id: 
-    :param hpo_id: 
-    :return: 
-    """
-    return bq_utils.table_exists(
-        bq_utils.get_table_id(hpo_id, PARTICIPANT_MATCH), ehr_dataset_id)
-
-
-def exist_identity_match(client, table_id):
-    """
-    This function checks if the hpo has valid the identity_match table
-
-    :param client: a BigQueryClient
-    :param table_id:
-    :return:
-    """
-    try:
-        client.get_table(table_id)
-        return True
-    except NotFound:
-        return False
-
-
-def get_missing_criterion(field_names):
-    """
-    This function generates a bigquery column expression for missing criteria
-    
-    :param field_names: a list of field names for counting `missing`s 
-    :return: 
-    """
-    joined_column_expr = ' + '.join([
-        CAST_MISSING_COLUMN.format(column=field_name)
-        for field_name in field_names
-    ])
-    return joined_column_expr
+SELECT_NOT_VALIDATED_PARTICIPANTS_QUERY = JINJA_ENV.from_string("""
+SELECT person_id
+FROM `{{project_id}}.{{ehr_dataset_id}}.{{participant_match_table}}`
+WHERE LOWER(algorithm_validation) != 'yes'
+AND LOWER(manual_validation) != 'yes'
+""")
 
 
-def get_list_non_match_participants(client, validation_dataset_id, hpo_id):
+class RemoveNonMatchingParticipant(BaseCleaningRule):
     """
-    This function retrieves a list of non-match participants
-
-    :param client: a BigQueryClient
-    :param validation_dataset_id:
-    :param hpo_id: 
-    :return: 
+    Removes records with person_ids that are not validated by sites and non-matching.
     """
 
-    # get the the hpo specific <hpo_id>_identity_match
-    identity_match_table = bq_utils.get_table_id(hpo_id, IDENTITY_MATCH)
-    result = []
-    fq_identity_match_table = f'{client.project}.{validation_dataset_id}.{identity_match_table}'
-    if not exist_identity_match(client, fq_identity_match_table):
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id,
+                 ehr_dataset_id, validation_dataset_id):
```

Comment:
I think `table_namer=None` should be a parameter here passed to super `__init__`.